plone/session/plugins/session.py

```diff
@@ -259,6 +294,23 @@ def updateCredentials(self, request, response, login, new_password):
 
     # ICredentialsResetPlugin implementation
     def resetCredentials(self, request, response):
+        if self.per_user_keyring:
+            # Prevent plone.protect from redirecting to @@confirm-action
+            # instead of logging-out.
+            # Caused by invalidating the user keyring.
+            alsoProvides(request, IDisableCSRFProtection)
```

Comment:
I do not like the overall catching `alsoProvides`. 

Is there any way to  use `safeWrite` here? https://github.com/plone/plone.protect/#if-you-just-want-to-allow-an-object-to-be-writable-on-a-request

Reply:
I quickly attempted `safeWrite` on `manager` and the portal object itself before opting to use `IDisableCSRFProtection` but it did not work, perhaps setting `safeWrite` on both would work, will try.

Reply:
I think you would explicitly do a safewrite on the keyring itself.

Reply:
Yes, that's what I did before opting for `IDisableCSRFProtection`.
But the strange thing is now that I pulled the latest https://github.com/plone/buildout.coredev (5.2 branch: b1919ae1f) it works without `safeWrite` or `IDisableCSRFProtection`, at least on my machine. Waiting on Jenkins.

Reply:
Now I'm extra confused, it appears that neither `safeWrite` or `IDisableCSRFProtection` are needed anymore. But Plone 6 on Python 3.7 fails (I think unrelated to my changes) while 3.8 and 3.9 work.

Reply:
> But Plone 6 on Python 3.7 fails (I think unrelated to my changes) while 3.8 and 3.9 work.

Looks unrelated indeed. I have restarted this job for good measure.

Reply:
So I guess everything works now. The plone.protect disables have been removed and tests pass (I only encountered the _@@confirm-action_ redirect in tests only).

Safe to merge?